packages/nextjs/utils/scaffold-eth/contract.ts

```diff
@@ -335,6 +337,68 @@ export type UseScaffoldEventHistoryData<
 
 export type AbiParameterTuple = Extract<AbiParameter, { type: "tuple" | `tuple[${string}]` }>;
 
+/**
+ * Enhanced error parsing that creates a lookup table from all deployed contracts
+ * to decode error signatures from any contract in the system
+ */
+const getParsedErrorWithAllAbis = (error: any, chainId: AllowedChainIds): string => {
+  const originalParsedError = getParsedError(error);
+
+  // Check if this is an unrecognized error signature
+  if (originalParsedError.includes("Encoded error signature") && originalParsedError.includes("not found on ABI")) {
+    const signatureMatch = originalParsedError.match(/0x[a-fA-F0-9]{8}/);
+    const signature = signatureMatch ? signatureMatch[0] : "";
+
+    if (signature) {
```

Comment:
Maybe we could early returns instead of nested if blocks 

if (!signature) return originalParsedError;

Reply:
[9f2e17e](https://github.com/scaffold-eth/scaffold-eth-2/pull/1146/commits/9f2e17e35e9e7de24d9ce4687c800642b3b6078d)